Modifications and scripts to build on tulip

mewall opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/lanl/bml/447)
<!-- Reviewable:end -->

nicolasbock commented on src/C-interface/dense/bml_diagonalize_dense.c:
```diff
@@ -333,9 +334,15 @@ bml_diagonalize_dense_single_complex(
+        float vu = FLT_MAX;
+        int il = 1;
+        int iu = A->N;
+        C_CHEEVR("V", "A", "U", &A->N, A_copy, &A->N, &vl, &vu, &il, &iu,
```
Why do we need the 4 parameters `vi`, `vu`, `il`, `iu`? According to the documentation for `CHEEVR` [1] those are only used if the `RANGE` parameter (the second one) is `V` or `I`.

[1] http://www.netlib.org/lapack/explore-html/d9/de3/group__complex_h_eeigen_gac02f8a55f35ec7767a36b5a9f3ebb0d8.html

mewall replied:
Sorry for the delay, I didn't receive an email notification about the response.

This change is required for compatibility with Cray LibSci. Passing NULL throws a SIGSEGV when calling zheevr_  or cheevr_. We just need to define something sensible for compatibility, it should be harmless otherwise. Thanks.

mewall replied:
@nicolasbock Is this issue resolved? Any remaining issues to be addressed?

nicolasbock replied:
Thanks for the information @mewall . Yes, that explains it.

nicolasbock reviewed (changes requested):
Please have a look also at the failed CI job. The linter isn't happy :smile:

jeanlucf22 commented on scripts/build_tulip_magma_cce.sh:
```diff
@@ -0,0 +1,30 @@
+export CMAKE_Fortran_FLAGS=${CMAKE_Fortran_FLAGS:="-L${CRAY_LIBSCI_PREFIX}/lib -L${HIP_PATH}/lib -L${MAGMA_HOME}/lib -L${ROCM_PATH}/lib -lsci_cray -lmagma -D__HIP_PLATFORM_HCC__ -DHAVE_HIP -ef -DCRAY_SDK"}
+export EXTRA_CFLAGS=${EXTRA_CFLAGS:="-I${CRAY_LIBSCI_PREFIX}/include -I${HIP_PATH}/include -I${MAGMA_HOME}/include -I${ROCM_PATH}/include -DHAVE_HIP -D__HIP_PLATFORM_HCC__ -DCRAY_SDK"}
+#export EXTRA_LINK_FLAGS=${EXTRA_LINK_FLAGS:="-L${HIP_PATH}/lib -L/home/users/coe0209/packages/openblas -lamdhip64 -lopenblas -lm"}
+export EXTRA_LINK_FLAGS=${EXTRA_LINK_FLAGS:="-L${CRAY_LIBSCI_PREFIX}/lib -L${HIP_PATH}/lib -L${MAGMA_HOME}/lib -L${ROCM_PATH}/lib -lsci_cray -lmagma"}
```
You should not need these extra flags for magma if you specify the environment variable MAGMA_ROOT. I am actually not sure if you need any of these extra flags...

mewall replied:
Your suspicions were correct, the flags have been simplified in the amended the commit.

mewall commented:
Fixed and rebased on master.


It looks like the only test that fails now is one that isn't associated with this pull request.

________________________________
From: Nicolas Bock ***@***.***>
Sent: Wednesday, March 24, 2021 1:23:33 AM
To: lanl/bml
Cc: Wall, Michael E; Author
Subject: [EXTERNAL] Re: [lanl/bml] Modifications and scripts to build on tulip (#447)


@nicolasbock requested changes on this pull request.

Please have a look also at the failed CI job. The linter isn't happy 😄

________________________________

In src/C-interface/dense/bml_diagonalize_dense.c<https://github.com/lanl/bml/pull/447#discussion_r599515578>:

> @@ -333,9 +334,15 @@ bml_diagonalize_dense_single_complex(

 #ifdef NOBLAS

     LOG_ERROR("No BLAS library");

 #else

-    C_CHEEVR("V", "A", "U", &A->N, A_copy, &A->N, NULL, NULL, NULL, NULL,

-             &abstol, &M, evals, evecs, &A->N, isuppz, work, &lwork, rwork,

-             &lrwork, iwork, &liwork, &info);

+    {

+        float vl = -FLT_MAX;

+        float vu = FLT_MAX;

+        int il = 1;

+        int iu = A->N;

+        C_CHEEVR("V", "A", "U", &A->N, A_copy, &A->N, &vl, &vu, &il, &iu,



Why do we need the 4 parameters vi, vu, il, iu? According to the documentation for CHEEVR [1] those are only used if the RANGE parameter (the second one) is V or I.

[1] http://www.netlib.org/lapack/explore-html/d9/de3/group__complex_h_eeigen_gac02f8a55f35ec7767a36b5a9f3ebb0d8.html

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lanl/bml/pull/447#pullrequestreview-618518226>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AA67VEJZIKU55QCP64UV37DTFCB4LANCNFSM4ZUJMNYQ>.

nicolasbock commented:
@jeanlucf22 Could you have a quick look? If you are ok with this change we can merge it.

jeanlucf22 commented:
@nicolasbock I am ok with this change

nicolasbock commented:
Thanks @jeanlucf22 !
